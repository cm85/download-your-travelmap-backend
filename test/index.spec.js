const AWS = require('aws-sdk-mock');

process.env.BUCKET = require('../infrastructure/variables.tf.json').variable[1].bucket.default;

let app;

beforeEach(() => {
  AWS.mock('S3', 'putObject');
  app = require('../app/index'); /* eslint-disable-line global-require */
});


// const app = require('../dist/app');

const cb = (nth, data) => ({
  body: JSON.parse(data.body),
  status: data.statusCode,
});

test('follows redirect', async () => {
  const result = await app.handler(
    { queryStringParameters: { url: 'http://www.tripadvisor.co.uk/MemberProfile-a_uid.F3B46B68117496775EE93A2AB6A9C1DC' } }, null,
    cb,
  );

  expect(result.body.username).toMatch('Atanas_GK');
});

test('test trim', async () => {
  const result = await app.handler(
    { queryStringParameters: { url: ' http://www.tripadvisor.com/members/christianhaller' } }, null,
    cb,
  );

  expect(result.body.username).toMatch('christianhaller');
});

test('map url', async () => {
  const result = await app.handler({ queryStringParameters: { url: 'https://www.tripadvisor.com/TravelMap-a_uid.F16A76DBDC7075B786CC2C71B9198693' } }, null, cb);

  expect(result.body.username).toMatch('christianhaller');
});

test('no profile url', async () => {
  const result = await app.handler({ queryStringParameters: { url: 'http://www.tripadvisor.co.uk' } }, null, cb);

  expect(result).toMatchSnapshot();
});

test('no tripadvisor  url', async () => {
  const result = await app.handler({ queryStringParameters: { url: 'http://christianhaller.com' } }, null, cb);

  expect(result).toMatchSnapshot();
});

test('no profile url', async () => {
  const result = await app.handler(
    { queryStringParameters: { url: 'https://www.tripadvisor.com.br/Saves?v=list#39842494' } }, null,
    cb,
  );

  expect(result).toMatchSnapshot();
});

test('no url', async () => {
  const result = await app.handler({ queryStringParameters: { url: 'x' } }, null, cb);

  expect(result).toMatchSnapshot();
});
