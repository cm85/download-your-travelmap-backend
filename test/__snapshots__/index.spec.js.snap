// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no profile url 1`] = `
Object {
  "body": "invalid Url",
  "status": 500,
}
`;

exports[`no profile url 2`] = `
Object {
  "body": "invalid Url",
  "status": 500,
}
`;

exports[`no tripadvisor  url 1`] = `
Object {
  "body": "Please enter your TripAdvisor profile URL",
  "status": 500,
}
`;

exports[`no url 1`] = `
Object {
  "body": "Please enter a valid URL",
  "status": 500,
}
`;
